### PR TITLE
chore: Improve chart data API + schemas + tests

### PR DESCRIPTION
**Pull request by villebro**
### CATEGORY

Choose one

- [x] Bug Fix
- [x] Enhancement (new features, refinement)
- [x] Refactor
- [x] Add tests
- [ ] Build / Development Environment
- [x] Documentation

### SUMMARY
Several small changes to chart data API/schema, `QueryContext`/`QueryObject`:
- Add query context related fixtures. Might seem overkill for now, but I intend to add more fixtures as we roll out the API to chart plugins.
- Move query context tests from `core_tests.py` to new `query_context_tests.py` (more to come)
- Change enums in chart api Marshmallow specs from `enum` to `validate=validate.OneOf()` to add validation (`enum` wasn't validating values) while still retaining `enum` in OpenAPI spec.

Related ticket: #9187 

### TEST PLAN
CI + local testing

### REVIEWERS
@dpgaspar

### CODE REVIEW

File: tests/base_tests.py [hunk 1]
```diff
@@ -127,21 +129,25 @@ def login(self, username="admin", password="general"):
...
         return db.session.query(Database).filter_by(id=db_id).one()
 
-    def get_druid_ds_by_name(self, name):
+    @staticmethod
+    def get_druid_ds_by_name(name: str) -> DruidDatasource:
         return db.session.query(DruidDatasource).filter_by(datasource_name=name).first()
 
-    def get_datasource_mock(self):
+    @staticmethod
+    def get_datasource_mock():
```
Comment (dpgaspar) [inline tests/base_tests.py:150]: nit: Missing return type annotation

File: tests/query_context_tests.py [hunk 2]
```diff
@@ -0,0 +1,94 @@
+# Licensed to the Apache Software Foundation (ASF) under one
```
Comment (dpgaspar) [inline tests/query_context_tests.py:1]: Would it make sense to place this on /tests/charts/query_context_tests ?

**Review by dpgaspar — APPROVED** [review]
Looks good, just some comments

File: tests/query_context_tests.py (hunk 2 above)
Comment (villebro) [reply tests/query_context_tests.py:1]: These tests are for `query_context.py` and `query_object.py` in `superset/common`, and in theory are not directly related to `tests/charts`.

Comment (dpgaspar) [reply tests/query_context_tests.py:1]: ok

**Pull request merged**